cmd/cosign/cli/attach/sig.go

```diff
@@ -95,9 +96,21 @@ func SignatureCmd(ctx context.Context, regOpts options.RegistryOptions, sigRef,
 			return err
 		}
 	}
-	bundle := bundle.TimestampToRFC3161Timestamp(timeStampedSig)
+	TSBundle := bundle.TimestampToRFC3161Timestamp(timeStampedSig)
 
-	newSig, err := mutate.Signature(sig, mutate.WithCertChain(cert, certChain), mutate.WithRFC3161Timestamp(bundle))
+	if rekorBundleRef != "" {
+		rekorBundleByte, err := os.ReadFile(filepath.Clean(rekorBundleRef))
+		if err != nil {
+			return err
+		}
+
+		rekorBundle, err = bundle.BytesToRekorBundle(rekorBundleByte)
```

Comment:
This should also sync up with the other PR - If you're writing a LocalSignedPayload, you should unmarshal to it before extracting the RekorBundle struct.